### PR TITLE
[Backport release-1.31] Bump containerd to v1.7.24

### PR DESCRIPTION
**Pull request by twz123**
Backport to `release-1.31`:

* #5314

### CODE REVIEW

File: cmd/controller/controller.go
Comment (jnummelin) [inline cmd/controller/controller.go:1]: These seem bit unrelated changes?

Comment (twz123) [reply cmd/controller/controller.go:1]: huh, how come... :open_mouth:  I didn't even look at the diff, after the usual cherry-pick :see_no_evil:

Comment (twz123) [reply cmd/controller/controller.go:1]: Thanks for spotting :roll_eyes:

**Review by jnummelin — APPROVED** [review]
